arraycontainer.go

```diff
@@ -62,6 +62,58 @@ func (ac *arrayContainer) getManyIterator() manyIterable {
 	return &shortIterator{ac.content, 0}
 }
 
+type arrayContainerUnsetIterator struct {
```

Comment:
So, in arraycontainer, I think we don't define iterator types, they are in shortiterator.go. I think it would be more consistent to move the code there, with the other iterators.

(Of course, this changes nothing with respect to the functionality.)